Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -230,14 +266,288 @@ ContourExtractor2DImageFilter<TInputImage>::GenerateData()
+    const IndexType     max{ bbox.second };
+    // Set boundary values in largerRegion to be distinct from label if they will be looked at.
+    if (min[0] == left_top[0])
+    {
+      const IndexType        stripeIndex{ min[0] - 1, min[1] - 1 };
+      const SizeType         stripeSize{ 1, static_cast<SizeValueType>(max[1] - min[1] + 3) };
+      const InputRegionType  stripeRegion{ stripeIndex, stripeSize };
+      const ImageRegionRange stripeRange{ *largerImage, stripeRegion };
+      const InputPixelType   differentLabel = label + 1;
+      std::fill(stripeRange.begin(), stripeRange.end(), differentLabel);
+    }
+    if (min[1] == left_top[1])
+    {
+      const IndexType        stripeIndex{ min[0] - 1, min[1] - 1 };
+      const SizeType         stripeSize{ static_cast<SizeValueType>(max[0] - min[0] + 3), 1 };
+      const InputRegionType  stripeRegion{ stripeIndex, stripeSize };
+      const ImageRegionRange stripeRange{ *largerImage, stripeRegion };
+      const InputPixelType   differentLabel = label + 1;
+      std::fill(stripeRange.begin(), stripeRange.end(), differentLabel);
+    }
+    if (max[0] == right_bot[0])
+    {
+      const IndexType        stripeIndex{ max[0] + 1, min[1] - 1 };
+      const SizeType         stripeSize{ 1, static_cast<SizeValueType>(max[1] - min[1] + 3) };
+      const InputRegionType  stripeRegion{ stripeIndex, stripeSize };
+      const ImageRegionRange stripeRange{ *largerImage, stripeRegion };
+      const InputPixelType   differentLabel = label + 1;
+      std::fill(stripeRange.begin(), stripeRange.end(), differentLabel);
+    }
+    if (max[1] == right_bot[1])
+    {
+      const IndexType        stripeIndex{ min[0] - 1, max[1] + 1 };
+      const SizeType         stripeSize{ static_cast<SizeValueType>(max[0] - min[0] + 3), 1 };
+      const InputRegionType  stripeRegion{ stripeIndex, stripeSize };
+      const ImageRegionRange stripeRange{ *largerImage, stripeRegion };
+      const InputPixelType   differentLabel = label + 1;
+      std::fill(stripeRange.begin(), stripeRange.end(), differentLabel);
+    }
+    const IndexType       shrunkIndex{ min[0] - 1, min[1] - 1 };
+    const SizeType        shrunkSize{ static_cast<SizeValueType>(max[0] - min[0]) + 2,
+                               static_cast<SizeValueType>(max[1] - min[1]) + 2 };
+    const InputRegionType shrunkRegion{ shrunkIndex, shrunkSize };
+
+    // A 1-pixel radius sets up a neighborhood with the following indices:
+    // 0 1 2
+    // 3 4 5
+    // 6 7 8
+    // We are interested only in the square of 4,5,7,8 which is the 2x2 square
+    // with the center pixel at the top-left. So we only activate the
+    // coresponding offsets, and only query pixels 4, 5, 7, and 8 with the
+    // iterator's GetPixel method.
+    SquareIterator it{ radius, largerImage, shrunkRegion };
+    it.ActivateOffset(none);
+    it.ActivateOffset(right);
+    it.ActivateOffset(down);
+    it.ActivateOffset(diag);
```

Comment:
It seems to me that these four `ActivateOffset` calls are unnecessary, because `GenerateDataForLabels()` does not use the "active index list" to iterate within a neighborhood. Right? I mean, do I understand correctly that these four `ActivateOffset` calls have no effect on the result?

Reply:
It's unclear to me, without diving into the code there.  The iterator that we are using there is nominally for a 3×3 square, but we only look at the lower 2×2 part of it.  In particular, the upper left corner can be outside the whole-image region by 1 pixel because the corresponding 2 by 2 will still be inside.  (Or if indexing is by the center of the 3×3 then the special case is not allowing the center to be within the image region at the rightmost or bottommost positions.)  Without making these ActivateOffset calls, I don't know how the iterator will know how to treat these edge cases.